src/Uri.php

```diff
@@ -275,7 +275,7 @@ public function withHost($host)
      */
     public function withPort($port)
     {
-        if (! (is_integer($port) || (is_string($port) && is_numeric($port)))) {
+        if (!is_numeric($port)) {
```

Comment:
I could have sworn `is_numeric()` did not return `true` for non-string values, but I just tested, and it does. Good change.